### PR TITLE
Allow user to verify and generate new recovery code

### LINKED ISSUE #62: Allow regenerating the recovery code
I recently lost the 12 words for a test device, and had to go to Settings to clear app data for Seedvault to reset the code.

It would be good to have an option in the menu for this, with one or more confirmations and warnings as needed.

- [ ] Allow regenrating passphrase
- [ ] Allow deleting old backups, per storage

### PR DESCRIPTION
**Pull request by grote**
https://user-images.githubusercontent.com/244947/105069500-ca295a00-5a60-11eb-9fe0-e5666a63a65c.mp4

Closes #62

### CODE REVIEW

File: app/src/main/res/drawable/ic_vpn_key.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (chirayudesai) [inline app/src/main/res/drawable/ic_vpn_key.xml:1]: ic_vpn_key ?

Missed renaming? :)

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -28,6 +28,8 @@
     <string name="settings_backup_status_summary">Last backup: %1$s</string>
     <string name="settings_backup_exclude_apps">Exclude apps</string>
     <string name="settings_backup_now">Backup now</string>
+    <string name="settings_backup_recovery_code">Recovery Code</string>
```
Comment (chirayudesai) [inline app/src/main/res/values/strings.xml:31]: This isn't related to this commit, but I think this is one the things that should be renamed.

Backup code / Backup passphrase / Backup key all sound better / clearer than Recovery code.

Can be it's own commit / PR, just bringing it up now since we're adding this new screen.

**Review by chirayudesai — COMMENTED** [review]
Some minor nits inline.

What does this look like right now, could you post the latest screenshot

File: app/src/main/java/com/stevesoltys/seedvault/crypto/KeyManager.kt [hunk 3]
```diff
@@ -47,7 +47,6 @@ internal class KeyManagerImpl : KeyManager {
 
     override fun storeBackupKey(seed: ByteArray) {
         if (seed.size < KEY_SIZE_BYTES) throw IllegalArgumentException()
-        // TODO check if using first 256 of 512 bits produced by PBKDF2WithHmacSHA512 is safe!
```
[the remark this answers is not in the document]
Comment (grote) [reply app/src/main/java/com/stevesoltys/seedvault/crypto/KeyManager.kt:50]: yep

File: app/src/main/res/drawable/ic_vpn_key.xml (hunk 1 above)
Comment (grote) [reply app/src/main/res/drawable/ic_vpn_key.xml:1]: No I am always leaving the official names for the files, so you can easily see if you add an icon that already exists.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (grote) [reply app/src/main/res/values/strings.xml:31]: Not sure I agree here. This is something for UX research.

File: app/src/main/java/com/stevesoltys/seedvault/settings/SettingsActivity.kt [hunk 4]
```diff
@@ -57,6 +58,9 @@ class SettingsActivity : RequireProvisioningActivity(), OnPreferenceStartFragmen
     ): Boolean {
         val fragment =
             supportFragmentManager.fragmentFactory.instantiate(classLoader, pref.fragment)
+        if (pref.key == "backup_recovery_code") fragment.arguments = Bundle().apply {
```
[the remark this answers is not in the document]
Comment (grote) [reply app/src/main/java/com/stevesoltys/seedvault/settings/SettingsActivity.kt]: pushed a new version that uses a variable for the pref.

**Comment by mikeNG** [comment]
Please don't use camel case for the new strings that are added

**Review by chirayudesai — APPROVED** [review]
Thanks for the video!

Current:
Verify | New

Alternatives:
New | Verify

Verify

New

The positioning is what I was talking about, but given the options we have there might not be a winner like we discussed last time, so we can just go with this for now and change later if need be.
